### PR TITLE
Feature / REL-1170 / Migration from sql to neo4j

### PR DESCRIPTION
**Pull request by vbyndych**
## Goal
As a part of effort to develop RDF persistence in neo4j, we need a procedure to migrate RDF data from SQL database (mysql or postgresql) to neo4j.

## Changelog
- feat: RDF migration from sql to neo4j 

## How to test / use

Internal use only.

### CODE REVIEW

File: core/kernel/api/class.ModelExporter.php [hunk 1]
```diff
@@ -29,29 +29,33 @@ class core_kernel_api_ModelExporter
     /**
      * Export the entire ontology
      *
+     * @param string $format (optional) which format resulted ontology will be.
+     *
      * @return string
      */
-    public static function exportAll()
+    public static function exportAll($format = 'rdfxml')
```
Comment (yaraslau-kavaliou) [inline core/kernel/api/class.ModelExporter.php:36]: Just a suggestion: maybe move 'rdfxml' to a constant?

Comment (vbyndych) [reply core/kernel/api/class.ModelExporter.php:36]: good catch. Will take a look.

File: core/kernel/api/class.ModelExporter.php [hunk 2]
```diff
@@ -68,20 +72,20 @@ public static function exportModelByUri()
...
-    private static function statement2rdf(PDOStatement $statement)
+    private static function statement2rdf($statement, $format = 'rdfxml')
     {
         $graph = new Graph();
         while ($r = $statement->fetch()) {
             if (isset($r['l_language']) && !empty($r['l_language'])) {
                 $graph->addLiteral($r['subject'], $r['predicate'], $r['object'], $r['l_language']);
             } elseif (common_Utils::isUri($r['object'])) {
-                $graph->add($r['subject'], $r['predicate'], $r['object']);
+                $graph->addResource($r['subject'], $r['predicate'], $r['object']);
```
[the remark this answers is not in the document]
Comment (vbyndych) [reply core/kernel/api/class.ModelExporter.php:82]: Yeah, this one really interesting. The thing is that `add` is a generic property/relation add method. However in this particular case, we are working with relation and we expect that object is a relation link, so it requires special treatment. But `add` method by default considers it as literal (string value) which results in incorrect output, so neo4j can't parse it properly. So if you go inside `getResource` you will see the following call:
```
$this->add(
            $resource,
            $property,
            array(
                'type' => substr($resource2, 0, 2) == '_:' ? 'bnode' : 'uri',
                'value' => $resource2
            )
```

**Review by augustas — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by yaraslau-kavaliou — APPROVED** [review]

**Pull request merged**